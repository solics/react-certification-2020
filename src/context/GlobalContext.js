import React, { createContext, useReducer } from 'react'
import { globalReducer, globalInitialState } from '../reducers'

const GlobalContext = createContext()

function GlobalContextProvider({ children }) {
	const [state, dispatch] = useReducer(globalReducer, globalInitialState)
	return (
		<GlobalContext.Provider value={[state, dispatch]}>{children}</GlobalContext.Provider>
	)
}

export { GlobalContext, GlobalContextProvider }
