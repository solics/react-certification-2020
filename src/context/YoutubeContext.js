import React, { createContext, useState } from 'react'
import { saveArrayOnLocalStorage } from '../utils/localStorage'

const API_KEY = process.env.REACT_APP_YT_API_KEY
const MAX_RESULTS = 20

const YoutubeContext = createContext()

const SEARCH_URL = `https://www.googleapis.com/youtube/v3/search?part=snippet&key=${API_KEY}&type=video&maxResults=${MAX_RESULTS}&`

function YoutubeContextProvider({ children }) {
	const [videos, setVideos] = useState([])
	const [loading, setLoading] = useState(false)
	const [loadingRelated, setLoadingRelated] = useState(false)
	const [errorMsg, setErrorMsg] = useState('')
	const [errorMsgRelated, setErrorMsgRelated] = useState('')
	const [videosRelated, setVideosRelated] = useState([])

	const search = term => {
		setLoading(true)
		fetch(`${SEARCH_URL}q=${term}`)
			.then(response => response.json())
			.then(data => {
				setLoading(false)
				setErrorMsg('')
				if (data?.items?.length) {
					setVideos(data.items)
					saveArrayOnLocalStorage('videos', data.items)
				} else {
					setVideos([])
					setErrorMsg(data.error.message)
				}
			})
			.catch(error => {
				setLoading(false)
				setErrorMsg(error.error.message)
				console.error(error)
			})
	}

	const getRelatedVideos = videoId => {
		setLoadingRelated(true)
		fetch(`${SEARCH_URL}relatedToVideoId=${videoId}`)
			.then(response => response.json())
			.then(data => {
				setLoadingRelated(false)
				setErrorMsgRelated('')
				if (data?.items?.length) {
					setVideosRelated(data.items)
				} else {
					setVideosRelated([])
					setErrorMsgRelated(data.error.message)
				}
			})
			.catch(error => {
				setLoadingRelated(false)
				setErrorMsgRelated(error.error.message)
				console.error(error)
			})
	}

	const defaultContext = {
		videos,
		videosRelated,
		search,
		setVideosContext: newVideos => {
			setVideos(newVideos)
		},
		getRelatedVideos,
		loading,
		loadingRelated,
		errorMsg,
		errorMsgRelated,
	}

	return (
		<YoutubeContext.Provider value={defaultContext}>{children}</YoutubeContext.Provider>
	)
}

export { YoutubeContext, YoutubeContextProvider }
