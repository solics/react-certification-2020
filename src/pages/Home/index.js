import React from 'react'
import styled from 'styled-components/macro'
import VideosList from '../../components/VideosList'

const HomeStyled = styled.div`
	padding: 20px;
`
const Title = styled.h1`
	text-align: center;
`

const Home = () => {
	return (
		<HomeStyled>
			<Title>Results</Title>
			<VideosList />
		</HomeStyled>
	)
}

export default Home
