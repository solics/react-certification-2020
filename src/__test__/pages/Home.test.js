import React from 'react'
import { render } from '@testing-library/react'
import { GlobalContextProvider } from '../../context/GlobalContext'
import Home from '../../pages/Home'

describe('Home', () => {
	it('Home Snapshot', () => {
		const component = render(
			<GlobalContextProvider>
				<Home />
			</GlobalContextProvider>
		)
		expect(component.container).toMatchSnapshot()
	})
})
