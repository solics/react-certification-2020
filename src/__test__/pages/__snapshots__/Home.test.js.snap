// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home Snapshot 1`] = `
<div>
  <div
    class="HomeStyled-sc-197bdb0-0 hhdQjV"
  >
    <h1
      class="Title-sc-197bdb0-1 kVCGVh"
    >
      Results
    </h1>
    <div
      class="VideosListStyled-sc-11o1z7z-0 bZFFBb"
    >
      <p
        class="MessageStyled-sc-11o1z7z-1 icPHbH"
      >
        There's no results
      </p>
    </div>
    
  </div>
</div>
`;
